yargs/index.d.ts

```diff
@@ -36,6 +36,12 @@ declare namespace yargs {
         demand(positionals: number, msg: string): Argv;
         demand(positionals: number, max: number, msg?: string): Argv;
```

Comment:
`demand()` is listed as deprecated in the docs with no hints on usage.  I vote we remove it here.